tests/integration/test_app_routes.py

```diff
@@ -1,139 +1,128 @@
-        yield client
-
-
-class StubResponse:
-
-    def __init__(self, fake_response_data=None):
-        self.fake_response_data = fake_response_data
 
-    def json(self):
-        return self.fake_response_data
+def single(iterator):
+    # Check for at least one item
+    try:
+        next(iterator)
+    except StopIteration:
+        return False
+    # Check for no more than one item
+    try:
+        next(iterator)
+    except StopIteration:
+        return True
+    else:
+        return False
 
-    def raise_for_status(self):
-        pass
 
+@pytest.fixture
+def client():
+    flaskenv_file_path = find_dotenv('.flaskenv.test')
+    load_dotenv(flaskenv_file_path)
 
-def request_stub(method, url, **kwargs):
-    test_board_id = os.environ.get('TRELLO_BOARD_ID')
-
-    # Board endpoints
-    if method == "GET" and url == f"{trello_api_base_url}/boards/{test_board_id}/lists":
-        return StubResponse(fake_response_data=[test_to_do_list, test_doing_list, test_done_list])
-    if method == "POST" and url == f"{trello_api_base_url}/boards/{test_board_id}/lists" \
-            and kwargs.get('data') is not None and 'name' in kwargs['data']:
-        return StubResponse(fake_response_data={**kwargs['data'], 'id': 'test-list-new'})
-
-    # List endpoints
-    for test_list in [test_to_do_list, test_doing_list, test_done_list]:
-        if method == "GET" and url == f"{trello_api_base_url}/lists/{test_list['id']}":
-            return StubResponse(fake_response_data=test_list)
-        if method == "GET" and url == f"{trello_api_base_url}/lists/{test_list['id']}/cards":
-            return StubResponse(fake_response_data=test_list['cards'])
-    if method == "GET" and url == f"{trello_api_base_url}/lists/test-list-new/cards":
-        return StubResponse(fake_response_data=[])
-
-    # Card endpoints
-    if method == "POST" and url == f"{trello_api_base_url}/cards" \
-            and kwargs.get('data') is not None and all(key in kwargs['data'] for key in ['name', 'desc', 'idList']):
-        return StubResponse(fake_response_data={**kwargs['data'], 'id': '031c3d1d3d902a6a89d20ae9'})
-    for test_card in [test_to_do_card, test_doing_card, test_done_card]:
-        if method == "GET" and url == f"{trello_api_base_url}/cards/{test_card['id']}":
-            return StubResponse(fake_response_data=test_card)
-        if method == "PUT" and url == f"{trello_api_base_url}/cards/{test_card['id']}" \
-                and kwargs.get('data') is not None and any(key in kwargs['data'] for key in ['name', 'desc', 'idList']):
-            return StubResponse(fake_response_data={**test_card, **kwargs['data']})
-        if method == "DELETE" and url == f"{trello_api_base_url}/cards/{test_card['id']}":
-            return StubResponse()
-
-    # Unhandled endpoints
-    raise NotImplementedError(f"Endpoint {url} not implemented for method {method} and data {kwargs.get('data')}")
+    from todo_app.app import create_app
```

Comment:
I don't like this! It works, but it's suggests a fairly fragile setup, and one that might get more fiddly as we add more tests

From looking it seems this is here so that the opening check in `app.py` trigger correctly, but I'd personally prefer just loading the relevant environment variables directly here. I could be convinced that I'm wrong, but I'm not sure I follow what the advantage of splitting out the `.env` files generally is, I see less duplication but it makes it much harder to follow which combo of environment variables is loading in each scenario, as opposed to:
* If I run the app normally (`poetry run flask run`), it uses `.env`
* If I run the tests, it uses `.env.test`
* If I use Docker, I tell it which one to use